units/task_stranded_cleanup.go

```diff
@@ -71,6 +74,19 @@ func (j *taskStrandedCleanupJob) Run(ctx context.Context) {
 		j.AddError(model.ClearAndResetStrandedTask(&h))
 	}
 
+	tasks, err := task.FindStuckDispatching()
+	if err != nil {
+		j.AddError(err)
+	}
+
+	for _, t := range tasks {
+		if time.Since(t.CreateTime) >= 2*7*24*time.Hour {
+			j.AddError(t.DeactivateTask(j.ID()))
+		} else {
+			j.AddError(model.TryResetTask(t.Id, evergreen.User, j.ID(), &t.Details))
```

Comment:
Why reset the task here?

Reply:
these are tasks which aren't running on a host, but stuck in the dispatching state. If we find one of those we should try and restart it rather than just letting it hang out forever?